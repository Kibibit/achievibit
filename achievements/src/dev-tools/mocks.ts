export class Shall {
  grantedAchievements: { [username: string]: any };

  grant(username, achievementObject) {
    this.grantedAchievements = this.grantedAchievements || {};
    this.grantedAchievements[username] = achievementObject;
  }
}

export class PullRequest {
  title = 'this is a happy little title';
  id = 'test';
  number: number;
  url = 'url';
  organization: { username: string };
  description = '';
  creator = {
    username: 'creator'
  };
  reviewers = [ {
    'username': 'reviewer'
  } ];
  merged: any;
  reviews: any;
  comments: any[];
  inlineComments: any[];
  reactions: any[];
  commits: any[];
  labels: string[];
  createdOn: Date;
  files: { name: string }[];
}
