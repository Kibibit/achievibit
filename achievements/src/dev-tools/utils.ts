export function createComment(username: string, message: string) {
  return {
    'author': {
      'username': username
    },
    'message': message
  };
}