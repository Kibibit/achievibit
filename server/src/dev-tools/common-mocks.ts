import { ArgumentsHost, Injectable } from '@nestjs/common';
import { InjectModel } from '@nestjs/mongoose';
import { prop as PersistInDb, ReturnModelType } from '@typegoose/typegoose';
import { Exclude, Expose } from 'class-transformer';

import { BaseModel, BaseService } from '@kb-abstracts';

export const mockResponse = {
  status: jest.fn().mockReturnThis(),
  json: jest.fn().mockReturnThis(),
  sendFile: jest.fn().mockReturnThis(),
  mockClear: () => {
    mockResponse.status.mockClear();
    mockResponse.json.mockClear();
    mockResponse.sendFile.mockClear();
  }
};

export const hostMock = (req, res, roles?: string[]): ArgumentsHost => {
  const ctx: any = {}
  ctx.switchToHttp = jest.fn().mockReturnValue({
      getRequest: () => req,
      getResponse: () => res
  });
  ctx.getHandler = jest.fn().mockReturnValue({ roles }) as Function;

  return ctx;
};

@Exclude()
export class MockModel extends BaseModel {
  @PersistInDb()
  mockPrivateAttribute: string;

  @Expose()
  @PersistInDb({ required: true })
  mockAttribute: string;

  @Exclude()
  @PersistInDb()
  updatedDate?: Date;

  constructor(partial: Partial<MockModel> = {}) {
    super();
    Object.assign(this, partial);
  }
}

@Injectable()
export class MockService extends BaseService<MockModel> {
  constructor(
    @InjectModel(MockModel.modelName)
    private readonly injectedModel: ReturnModelType<typeof MockModel>
  ) {
    super(injectedModel, MockModel);
  }
}
