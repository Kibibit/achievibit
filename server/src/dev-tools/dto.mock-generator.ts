import { Chance } from 'chance';
import { kebabCase, times } from 'lodash';
import { ObjectId } from 'mongodb';
import RandExp from 'randexp';

import { PullRequest, Repo, User } from '@kb-models';

type IAchievementFiles = Record<string, Record<string, unknown>>;

interface IChanceMixin {
  mongoObjectId: () => ObjectId;
  mongodbUrl: () => string;
  user: () => User;
  users: () => User[];
  repo: () => Repo;
  repos: () => Repo[];
  pullRequest: () => PullRequest;
  achievementScripts: (numOfAchievements?: number) => IAchievementFiles;
  randexp: (regex: RegExp) => RandExp;
}

const chance = new Chance() as Chance.Chance & IChanceMixin;

export const DtoMockGenerator: Chance.Chance & IChanceMixin =
  chance as Chance.Chance & IChanceMixin;

const customMixin: IChanceMixin & Chance.MixinDescriptor = {
  mongodbUrl(): string {
    return chance.url({ protocol: 'mongodb' + (chance.bool() ? '+srv' : '') });
  },
  randexp(regex: RegExp): RandExp {
    const randExp = new RandExp(regex);
    randExp.max = 10;

    return randExp;
  },
  mongoObjectId(): ObjectId {
    return new ObjectId(chance.string({
      pool: 'abcdefABCDEF123456789',
      length: 24
    }));
  },
  user(): User {
    const isOrg = chance.bool();

    return new User({
      _id: chance.mongoObjectId(),
      username: chance.name(),
      avatar: chance.url(),
      organization: isOrg,
      repos: times(chance.integer({ min: 0, max: 10 }), () => chance.animal()),
      url: chance.url(),
      users: isOrg ?
        times(chance.integer({ min: 0, max: 10 }), () => chance.animal()) :
        undefined,
      achievements: [],
      token: chance.string()
    });
  },
  users(): User[] {
    return times(chance.integer({ min: 0, max: 10 }), () => chance.user());
  },
  repo(): Repo {
    return new Repo({
      _id: chance.mongoObjectId(),
      name: chance.string(),
      fullname: chance.string(),
      organization: chance.company(),
      url: chance.url()
    });
  },
  repos(): Repo[] {
    return times(chance.integer({ min: 0, max: 10 }), () => chance.repo());
  },
  pullRequest() {
    return new PullRequest({
      createdOn: chance.date(),
      creator: chance.name(),
      description: chance.paragraph(),
      number: chance.integer(),
      prid: chance.guid(),
      url: chance.url()
    });
  },
  achievementScripts(
    numOfAchievements: number = chance.integer({ min: 1, max: 25 })
  ): IAchievementFiles {
    const scriptsObject = {};

    times(
      numOfAchievements,
      () => scriptsObject[ `${ kebabCase(chance.animal()) }.achievement` ] = {}
    );

    return scriptsObject;
  }
};

chance.mixin(customMixin);
