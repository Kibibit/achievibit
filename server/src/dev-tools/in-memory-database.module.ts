import { MongooseModule, MongooseModuleOptions } from '@nestjs/mongoose';
import { MongoMemoryServer } from 'mongodb-memory-server';
import mongoose from 'mongoose';

let mongod: MongoMemoryServer;

export const createInMemoryDatabaseModule =
  (options: MongooseModuleOptions = {}) => MongooseModule.forRootAsync({
    useFactory: async () => {
      mongod = new MongoMemoryServer();
      const mongoUri = await mongod.getUri();
      return {
        uri: mongoUri,
        ...options
      }
    }
  });

export const closeInMemoryDatabaseConnection = async () => {
  await mongoose.disconnect();
  if (mongod) await mongod.stop();
}
