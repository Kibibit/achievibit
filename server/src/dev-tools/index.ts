/**
 * @file Automatically generated by barrelsby.
 */

export * from './common-mocks';
export * from './dto.mock-generator';
export * from './in-memory-database.module';
export * from './captured-events/pull-request-assignee-added.event';
export * from './captured-events/pull-request-assignee-removed.event';
export * from './captured-events/pull-request-created-organization.event';
export * from './captured-events/pull-request-created.event';
export * from './captured-events/pull-request-edited.event';
export * from './captured-events/pull-request-label-added.event';
export * from './captured-events/pull-request-label-removed.event';
export * from './captured-events/pull-request-labels-initialized.event';
export * from './captured-events/pull-request-merged.event';
export * from './captured-events/pull-request-review-request-added.event';
export * from './captured-events/pull-request-review-request-removed.event';
export * from './captured-events/review-comment-added.event';
export * from './captured-events/review-comment-edited.event';
export * from './captured-events/review-comment-removed.event';
export * from './captured-events/review-submitted.event';
export * from './captured-events/wehbook-ping.event';
