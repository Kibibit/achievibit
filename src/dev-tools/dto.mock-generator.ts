import { Chance } from 'chance';
import { kebabCase, times } from 'lodash';
import { ObjectId } from 'mongodb';
import RandExp from 'randexp';

import { RepoDto, User } from '@kb-models';

interface IChanceMixin {
  mongoObjectId: () => ObjectId;
  mongodbUrl: () => string;
  user: () => User;
  users: () => User[];
  repoDto: () => RepoDto;
  repoDtos: () => RepoDto[];
  achievementScripts: (numOfAchievements?: number) => { [key: string]: {} };
  randexp: (regex: RegExp) => RandExp;
}

const chance = new Chance() as Chance.Chance & IChanceMixin;

export const DtoMockGenerator: Chance.Chance & IChanceMixin = chance as Chance.Chance & IChanceMixin;

const customMixin: IChanceMixin & Chance.MixinDescriptor = {
  mongodbUrl(): string {
    return chance.url({ protocol: 'mongodb' + (chance.bool() ? '+srv' : '') });
  },
  randexp(regex: RegExp): RandExp {
    const randExp = new RandExp(regex);
    randExp.max = 10;

    return randExp;
  },
  mongoObjectId(): ObjectId {
    return new ObjectId(chance.string({ pool: 'abcdefABCDEF123456789', length: 24 }));
  },
  user(): User {
    const isOrg = chance.bool();

    return new User({
      _id: chance.mongoObjectId(),
      __v: '0',
      username: chance.name(),
      avatar: chance.url(),
      organization: isOrg,
      repos: times(chance.integer({ min: 0, max: 10 }), () => chance.animal()),
      url: chance.url(),
      users: isOrg ?
        times(chance.integer({ min: 0, max: 10 }), () => chance.animal()) :
        undefined,
      achievements: [],
      token: chance.string()
    });
  },
  users(): User[] {
    return times(chance.integer({ min: 0, max: 10 }), () => chance.user());
  },
  repoDto(): RepoDto {
    return new RepoDto({
      _id: chance.mongoObjectId(),
      __v: 0,
      name: chance.string(),
      fullname: chance.string(),
      organization: chance.company(),
      url: chance.url()
    });
  },
  repoDtos(): RepoDto[] {
    return times(chance.integer({ min: 0, max: 10 }), () => chance.repoDto());
  },
  achievementScripts(numOfAchievements: number = chance.integer({ min: 1, max: 25 })): { [key: string]: {} } {
    const scriptsObject = {};

    times(numOfAchievements, () => scriptsObject[`${kebabCase(chance.animal())}.achievement`] = {});

    return scriptsObject;
  }
};

chance.mixin(customMixin);
