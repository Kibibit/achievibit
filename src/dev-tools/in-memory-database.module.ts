import { Module } from '@nestjs/common';
import { MongooseModule } from '@nestjs/mongoose';
import { MongoMemoryServer } from 'mongodb-memory-server';
import mongoose from 'mongoose';

let mongod: MongoMemoryServer;
let uri: string;

async function testDBFactory() {
  mongod = new MongoMemoryServer();
  uri = await mongod.getConnectionString();
  await mongoose.connect(uri, {
    useNewUrlParser: true,
    useUnifiedTopology: true
  });
  return { uri };
}

@Module({
  imports: [
    MongooseModule.forRootAsync({
      useFactory: testDBFactory
    })
  ]
})
export class InMemoryDatabaseModule {
  static async closeDatabase() {
    await mongoose.connection.dropDatabase();
    await mongoose.connection.close();
    await mongod.stop();
    mongod = null;
    uri = null;
  }
}
