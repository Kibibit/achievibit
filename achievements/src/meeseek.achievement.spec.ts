import { PullRequest, Shall } from './dev-tools/mocks';
import { meeseek } from './meeseek.achievement';

describe('meeseek achievement', () => {
  it('should not be granted if no issues resolved', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    meeseek.check(pullRequest, testShall);
    expect(testShall.grantedAchievements).toBeUndefined();
  });

  it('should not be granted if 1 issues resolved', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[resolves #1]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);
    expect(testShall.grantedAchievements).toBeUndefined();
  });

  it('should not be granted if 2 issues resolved', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[resolves #1]',
      '[resolves #2]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);
    expect(testShall.grantedAchievements).toBeUndefined();
  });

  it('should be granted to PR creator if 4 issues resolved', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[resolves #1]',
      '[resolves #2]',
      '[resolves #3]',
      '[resolves #4]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);

    expect(testShall.grantedAchievements.creator).toMatchSnapshot();
  });

  it('should not count same issue twice', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[resolves #1]',
      '[resolves #1]',
      '[resolves #1]',
      '[closes #1]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);
    expect(testShall.grantedAchievements).toBeUndefined();
  });

  it('should be case insensitive', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[Resolves #1]',
      '[resolves #2]',
      '[resolves #3]',
      '[resolves #4]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);

    expect(testShall.grantedAchievements.creator).toMatchSnapshot();
  });

  it('should not count if keyword is between other letters', () => {
    const testShall = new Shall();
    const pullRequest = new PullRequest();

    pullRequest.description = [
      '[awesomeresolves #1]', // should not be recognized
      '[resolves #2resolves #3]', // should not be recognized
      'resolves #3nice!',
      '[resolves #4]',
      '[resolves #5]',
      '[resolves #6]'
    ].join('\n');

    meeseek.check(pullRequest, testShall);
    expect(testShall.grantedAchievements).toBeUndefined();
  });

  describe('Github\'s resolve keywords', () => {
    it('should recognize "close"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[close #1]', // should not be recognized
        '[close #2]', // should not be recognized
        '[close #3]',
        '[close #4]',
        '[close #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "closes"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[closes #1]', // should not be recognized
        '[closes #2]', // should not be recognized
        '[closes #3]',
        '[closes #4]',
        '[closes #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "closed"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[closed #1]', // should not be recognized
        '[closed #2]', // should not be recognized
        '[closed #3]',
        '[closed #4]',
        '[closed #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "fix"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[fix #1]', // should not be recognized
        '[fix #2]', // should not be recognized
        '[fix #3]',
        '[fix #4]',
        '[fix #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "fixes"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[fixes #1]', // should not be recognized
        '[fixes #2]', // should not be recognized
        '[fixes #3]',
        '[fixes #4]',
        '[fixes #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "fixed"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[fixed #1]',
        '[fixed #2]',
        '[fixed #3]',
        '[fixed #4]',
        '[fixed #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "resolve"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[resolve #1]',
        '[resolve #2]',
        '[resolve #3]',
        '[resolve #4]',
        '[resolve #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);
      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "resolves"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[resolves #1]',
        '[resolves #2]',
        '[resolves #3]',
        '[resolves #4]',
        '[resolves #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize "resolved"', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[resolved #1]',
        '[resolved #2]',
        '[resolved #3]',
        '[resolved #4]',
        '[resolved #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);

      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
    it('should recognize a combination', () => {
      const testShall = new Shall();
      const pullRequest = new PullRequest();

      pullRequest.description = [
        '[closed #1]',
        '[close #2]',
        '[fixed #3]',
        '[resolves #4]',
        '[resolved #5]'
      ].join('\n');

      meeseek.check(pullRequest, testShall);
      expect(testShall.grantedAchievements.creator).toMatchSnapshot();
    });
  });
});
