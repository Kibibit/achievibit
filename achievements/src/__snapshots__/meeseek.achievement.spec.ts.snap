// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meeseek achievement Github's resolve keywords should recognize "close" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "closed" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "closes" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "fix" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "fixed" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "fixes" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "resolve" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "resolved" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize "resolves" 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement Github's resolve keywords should recognize a combination 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement should be case insensitive 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;

exports[`meeseek achievement should be granted to PR creator if 4 issues resolved 1`] = `
Object {
  "avatar": "images/achievements/meeseek.achievement.gif",
  "description": "<p>Congrats on resolving so many issues at ones! Shouldn't pull requests be kept simple?</p><p>Pull requests don't usually have to exist this long. It's getting weird.</p>",
  "name": "I'm Mr. Meeseeks! Look at me!",
  "relatedPullRequest": "test",
  "short": "Knock yourselves out. Just eh-keep your requests simple.",
}
`;
